### PR TITLE
3589 home screen

### PR DESCRIPTION
**Pull request by dinhquang251**
3589 create UI home screen using coordinatorLayout with toolbar and viewpager
https://edu-redmine.framgia.vn/issues/3589

### CODE REVIEW

**Comment by dinhquang251** [comment]
ready

File: app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import com.framgia.quangtran.music_42.ui.fragments.FragmentHome;
+import com.framgia.quangtran.music_42.ui.fragments.FragmentPersonal;
+import com.framgia.quangtran.music_42.ui.fragments.FragmentSetting;
+
+public class ViewPagerAdapter extends FragmentPagerAdapter {
+    private static final String HOME = "HOME";
+    private static final String PERSONAL = "PERSONAL";
+    private static final String SETTING = "SETTING";
+    private static final int SUMFRAGMENT= 3;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java]: format code

File: app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java]: remove bl

File: app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    public ViewPagerAdapter(FragmentManager fm) {
+        super(fm);
+    }
+
+    @Override
+    public Fragment getItem(int i) {
+        switch (i) {
+            case 0:
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java]: 26 28 30 hardcode.

File: app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                return new FragmentHome();
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java]: Đặt sai tên Fragment rồi e

File: app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    public int getCount() {
+        return SUMFRAGMENT;
+    }
+
+    @Nullable
+    @Override
+    public CharSequence getPageTitle(int position) {
+        String title = "";
+        switch (position) {
+            case 0:
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java]: 46 49 52 hardcode

File: app/src/main/java/com/framgia/quangtran/music_42/ui/fragments/FragmentHome.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.framgia.quangtran.music_42.R;
+
+public class FragmentHome extends Fragment {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/fragments/FragmentHome.java]:
```suggestion
public class HomeFragment extends Fragment {
```

File: app/src/main/java/com/framgia/quangtran/music_42/ui/fragments/FragmentHome.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_home, container, false);
+        return view;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/fragments/FragmentHome.java]:
```suggestion
        return inflater.inflate(R.layout.fragment_home, container, false);
```

File: app/src/main/java/com/framgia/quangtran/music_42/ui/fragments/FragmentPersonal.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.framgia.quangtran.music_42.R;
+
+public class FragmentPersonal extends Fragment {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/fragments/FragmentPersonal.java]:
```suggestion
public class PersonalFragment extends Fragment {
```

File: app/src/main/java/com/framgia/quangtran/music_42/ui/fragments/FragmentPersonal.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_personal, container, false);
+        return view;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/fragments/FragmentPersonal.java]:
```suggestion
        return inflater.inflate(R.layout.fragment_personal, container, false);
```

File: app/src/main/java/com/framgia/quangtran/music_42/ui/fragments/FragmentSetting.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.framgia.quangtran.music_42.R;
+
+public class FragmentSetting extends Fragment {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/fragments/FragmentSetting.java]:
```suggestion
public class SettingFragment extends Fragment {
```

File: app/src/main/java/com/framgia/quangtran/music_42/ui/fragments/FragmentSetting.java [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_setting, container, false);
+        return view;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/fragments/FragmentSetting.java]:
```suggestion
        return inflater.inflate(R.layout.fragment_setting, container, false);
```

File: app/src/main/res/layout/activity_home.xml [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+<android.support.design.widget.CoordinatorLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@android:color/background_light"
+    android:fitsSystemWindows="true">
+
+    <android.support.design.widget.AppBarLayout
+        android:id="@+id/appbar_ui"
```
Comment (NguyenMinhTuanB) [inline app/src/main/res/layout/activity_home.xml]:
```suggestion
        android:id="@+id/app_bar"
```

File: app/src/main/res/layout/activity_splash.xml [hunk 13]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout
```
Comment (NguyenMinhTuanB) [inline app/src/main/res/layout/activity_splash.xml]: FrameLayout là được

**Comment by NguyenMinhTuanB** [comment]
commented

**Comment by dinhquang251** [comment]
ready

File: app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+import android.support.v4.app.FragmentPagerAdapter;
+import com.framgia.quangtran.music_42.ui.fragments.HomeFragment;
+import com.framgia.quangtran.music_42.ui.fragments.PersonalFragment;
+import com.framgia.quangtran.music_42.ui.fragments.SettingFragment;
+
+public class ViewPagerAdapter extends FragmentPagerAdapter {
+    private static final String HOME = "HOME";
+    private static final String PERSONAL = "PERSONAL";
+    private static final String SETTING = "SETTING";
+    private static final int SUMFRAGMENT = 3;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java]:
```suggestion
    private static final int TOTAL_TABS = 3;
```

File: app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+    private static final int ONEFRAGMENT = 0;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java]:
```suggestion
    private static final int TAB_HOME = 0;
```

File: app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+    private static final int TWOFRAGMENT = 1;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java]:
```suggestion
    private static final int TAB_PERSONAL = 1;
```

File: app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+    private static final int THREEFRAGMENT = 2;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/framgia/quangtran/music_42/ui/adapters/ViewPagerAdapter.java]:
```suggestion
    private static final int TAB_SETTING = 2;
```

**Comment by dinhquang251** [comment]
ready

**Comment by dinhquang251** [comment]
ready

**Review by NguyenMinhTuanB — APPROVED** [review]

**Pull request merged**